pytest_bdd/steps.py

```diff
@@ -61,7 +59,7 @@ def get_step_fixture_name(name, type_, encoding=None):
     :rtype: string
     """
     return "pytestbdd_{type}_{name}".format(
-        type=type_, name=force_encode(name, **(dict(encoding=encoding) if encoding else {}))
+        type=type_, name=str(name).encode(**(dict(encoding=encoding) if encoding else {}))
     )
```

Comment:
force_encode(...) was always a noop in python3, but now the code actually does something different, it encodes it to bytes.
I think this should just be:
```suggestion
    return f"pytest_bdd_{type_}_{name}"
```

Reply:
When I took look at this code I've also thought so, but a lot of tests become red. So I've decided to not make refactoring this time but just leave old behavior.

Reply:
but this is not doing what the old code did. In python 3,`force_encode(obj)` always returned `obj` without changing anything.

Now it is converting the object to bytes, and then we string format that bytes object.

With the proposed change, the result will be like
`"pytest_bdd_given_b'there_is_a_foo'"`
instead of
`"pytest_bdd_given_there_is_a_foo"`